packages/eslint/src/migrations/update-17-2-0/move-options-to-target-defaults.ts

```diff
@@ -0,0 +1,184 @@
+    lintDefaults.inputs = [
+      'default',
+      ...(eslintConfig ? [`{workspaceRoot}/${eslintConfig}`] : []),
+      '{workspaceRoot}/tools/eslint-rules/**/*',
+    ];
+  }
+
+  // Cleanup old target defaults
+  const projects = graph.nodes;
+  const projectMap = getProjects(tree);
+
+  for (const [targetDefaultKey, targetDefault] of Object.entries(
+    nxJson.targetDefaults
+  )) {
+    if (
+      !isTargetDefaultUsed(
+        targetDefault,
+        nxJson.targetDefaults,
+        projects,
+        projectMap
+      )
+    ) {
+      delete nxJson.targetDefaults[targetDefaultKey];
+    }
+  }
+
+  let addOutputs = false;
+  forEachExecutorOptions<Schema>(
+    tree,
+    '@nx/eslint:lint',
+    (options, proj, targetName, configuration) => {
+      const projConfig = projectMap.get(proj);
+
+      // modify lint file patterns
+      if (options.lintFilePatterns) {
+        const projectRootRegex = new RegExp(`^${projConfig.root}/`);
+        projConfig.targets[targetName].options.lintFilePatterns =
+          options.lintFilePatterns
+            .filter(
+              (filePattern) =>
+                filePattern !== '{projectRoot}' &&
+                filePattern !== projConfig.root
+            )
+            .map((filePattern) =>
+              filePattern.replace(projectRootRegex, '{projectRoot}/')
+            );
+        // remove lintFilePatterns if empty
+        if (
+          projConfig.targets[targetName].options.lintFilePatterns.length === 0
+        ) {
+          delete projConfig.targets[targetName].options.lintFilePatterns;
+        }
+      }
+
+      // remove inputs if they are not bringing any new inputs
+      if (
+        projConfig.targets[targetName].inputs &&
+        projConfig.targets[targetName].inputs.every((i) =>
+          lintDefaults.inputs.includes(i)
+        )
+      ) {
+        delete projConfig.targets[targetName].inputs;
+      }
+
+      // remove obsolete eslint config definition, unless it's a custom one
+      const projectEslintConfig = findEslintFile(tree, projConfig.root);
+      if (
+        options.eslintConfig === `${projConfig.root}/${projectEslintConfig}` ||
+        options.eslintConfig === `{projectRoot}/${projectEslintConfig}`
+      ) {
+        delete projConfig.targets[targetName].options.eslintConfig;
+      }
+
+      // remove options if empty
+      if (Object.keys(projConfig.targets[targetName].options).length === 0) {
+        delete projConfig.targets[targetName].options;
+      }
+
+      // track output
+      if (options.outputFile) {
+        addOutputs = true;
+      }
+      if (
+        projConfig.targets[targetName].outputs?.length === 1 &&
+        projConfig.targets[targetName].outputs[0] === '{options.outputFile}'
+      ) {
+        delete projConfig.targets[targetName].outputs;
+      }
+      updateProjectConfiguration(tree, proj, projConfig);
+    }
+  );
+
+  if (addOutputs) {
+    lintDefaults.outputs = ['{options.outputFile}'];
```

Comment:
why are we adding outputs, though? if they are already there, we want to add them, but in new generated projects we don't want them?

Reply:
Currently, we always set `outputs,` but it's almost never used, so it ends up being just a boilerplate code.
With this fix, we always remove it from `project.json`, and we only add it to targetDefaults if one of the projects is defining outputFile.